packages/components/src/utils/browser.ts

```diff
@@ -1,10 +1,18 @@
 type Browser = 'Safari' | 'others'
 
-const isSafari = /^((?!chrome|android).)*safari/i.test(navigator.userAgent)
+const isSafari = /^((?!chrome|android).)*safari/i.test(getUserAgent())
 
 export const isBrowser = (browser: Browser): boolean => {
   if (browser === 'Safari') {
     return isSafari
   }
   return false
 }
+
+function getUserAgent(): string {
+  if (typeof (window as any) !== 'undefined' && window?.navigator && navigator?.userAgent) {
+    return navigator.userAgent
```

Comment:
maybe you can write something like this: 
`    return navigator.userAgent ?? '' `